internal/state/execution.go

```diff
@@ -414,43 +411,47 @@ func execBlockOnProxyApp(
 	return abciResponses, nil
 }
 
-func getBeginBlockValidatorInfo(block *types.Block, store Store,
-	initialHeight int64) abci.LastCommitInfo {
-	voteInfos := make([]abci.VoteInfo, block.LastCommit.Size())
-	// Initial block -> LastCommitInfo.Votes are empty.
-	// Remember that the first LastCommit is intentionally empty, so it makes
-	// sense for LastCommitInfo.Votes to also be empty.
-	if block.Height > initialHeight {
-		lastValSet, err := store.LoadValidators(block.Height - 1)
-		if err != nil {
-			panic(err)
+func buildLastCommitInfo(block *types.Block, store Store, initialHeight int64) abci.LastCommitInfo {
+	if block.Height == initialHeight {
+		// there is no last commmit for the initial height.
+		// return an empty value.
+		return abci.LastCommitInfo{
+			Round: 0,
+			Votes: make([]abci.VoteInfo, 0),
```

Comment:
Do we need to explicitly allocate anything here? If we just want length zero, this could all be

```go
return abci.LastCommitInfo{}
```

Reply:
Good point. I was paranoid about serializers barfing on the nil pointer, but perhaps I was being too cautious.